std/path.d

```diff
@@ -757,21 +757,60 @@ unittest
 
 
 /*  Helper function that returns the position of the filename/extension
-    separator dot in path.  If not found, returns -1.
+    separator dot in path.
+
+    Params:
+        path = file spec as string or indexable range
+    Returns:
+        index of extension separator (the dot), or -1 if not found
 */
 private ptrdiff_t extSeparatorPos(R)(const R path)
-    if (isRandomAccessRange!R && hasSlicing!R && isSomeChar!(ElementType!R) ||
+    if (isRandomAccessRange!R && hasLength!R && isSomeChar!(ElementType!R) ||
         isNarrowString!R)
 {
-    auto i = (cast(ptrdiff_t) path.length) - 1;
-    while (i >= 0 && !isSeparator(path[i]))
+    for (auto i = cast(ptrdiff_t)path.length - 1;
+         i >= 0 && !isSeparator(path[i]);
+         --i)
```

Comment:
The C "downto" idiom works nicely here:

    for (auto i = path.length; i-- > 0 && !isSeparator(path[i]); )